mongoex/system.go

```diff
@@ -18,6 +18,8 @@ type Config struct {
 	UseTLS  bool
 }
 
+// Load connects to mongo. The context passed in is expected to carry an o11y provider
+// and is only used for reporting (not for cancellation),
```

Comment:
In the spirit of not forcing all consumers of this library to use `system`, we should split out a `New` function, like we have for the other helper packages, that allows you just to wire in `o11y`, but not use `system`.

(non blocking for this PR, just think it's the right thing to do)